### PR TITLE
[4.0] Refactor JClasses in com_users

### PR DESCRIPTION
**Pull request by alexandraciobica**
### Summary of Changes

As part of gsoc2018 Enhance Users project, I refactored all the JClasses that could be imported using namespaces, except JForm because I should have refactored also the parent.

### CODE REVIEW

**Comment by dgrammatiko** [comment]
Can you please make this PR against the main repo/ branch 4.0-dev?

**Comment by bembelimen** [comment]
Hi @dgrammatiko isn't this the case?

**Comment by dgrammatiko** [comment]
@bembelimen yup, somehow I was confused checking the gsoc projects 🤷‍♂️

File: administrator/components/com_users/Controller/LevelController.php [hunk 1]
```diff
@@ -13,6 +13,9 @@
 use Joomla\CMS\Access\Exception\Notallowed;
 use Joomla\CMS\MVC\Controller\FormController;
 use Joomla\Utilities\ArrayHelper;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Session\Session;
+
```
Comment (Quy) [inline administrator/components/com_users/Controller/LevelController.php]: Remove blank line.

File: administrator/components/com_users/Controller/MailController.php [hunk 2]
```diff
@@ -11,6 +11,10 @@
 defined('_JEXEC') or die;
 
 use Joomla\CMS\MVC\Controller\BaseController;
+use Joomla\CMS\Router\Route;
+use Joomla\CMS\Session\Session;
+use Joomla\CMS\Language\Text;
+
```
Comment (Quy) [inline administrator/components/com_users/Controller/MailController.php]: Remove blank line

File: administrator/components/com_users/Model/DebuguserModel.php [hunk 3]
```diff
@@ -115,10 +118,11 @@ public function getItems()
 	 * @return  void
 	 *
 	 * @since   1.6
+	 * @throws \Exception
```
Comment (Quy) [inline administrator/components/com_users/Model/DebuguserModel.php]: Add space before \

File: administrator/components/com_users/Model/GroupModel.php [hunk 4]
```diff
@@ -96,11 +98,12 @@ public function getForm($data = array(), $loadData = true)
 	 * @return  mixed  The data for the form.
 	 *
 	 * @since   1.6
+	 * @throws \Exception
```
Comment (Quy) [inline administrator/components/com_users/Model/GroupModel.php]: Add space before \

File: administrator/components/com_users/Model/LevelModel.php [hunk 5]
```diff
@@ -168,11 +171,12 @@ public function getForm($data = array(), $loadData = true)
 	 * @return  mixed  The data for the form.
 	 *
 	 * @since   1.6
+	 * @throws \Exception
```
Comment (Quy) [inline administrator/components/com_users/Model/LevelModel.php]: Add space before \

File: administrator/components/com_users/tmpl/groups/default.php [hunk 6]
```diff
@@ -9,50 +9,58 @@
...
 defined('_JEXEC') or die;
 
+use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Router\Route;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Layout\LayoutHelper;
+use Joomla\CMS\Access\Access;
+
+
```
Comment (Quy) [inline administrator/components/com_users/tmpl/groups/default.php]: Remove blank line

File: components/com_users/Model/ResetModel.php [hunk 7]
```diff
@@ -12,6 +12,13 @@
 
 use Joomla\CMS\Application\ApplicationHelper;
 use Joomla\CMS\MVC\Model\FormModel;
+use Joomla\CMS\Factory;
+use Joomla\CMS\User\User;
+use Joomla\CMS\String\PunycodeHelper;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Router\Route;
+use Joomla\CMS\User\UserHelper;
+
```
Comment (Quy) [inline components/com_users/Model/ResetModel.php]: Remove blank line

**Comment by alexandraciobica** [comment]
Thank you @Quy for the feedback! I did the fixes.

**Comment by laoneo** [comment]
Thank you very much for this one. We will merge it soon into our GsoC repo.

**Pull request merged**